enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md

```diff
@@ -0,0 +1,375 @@
+>can use as is.  But do we want to take on adding such a feature upstream to that repo? **This question most
+>likely can be deferred until implementation.**
+> 2. What will be the precise means of interacting with the host filesystem.  The hostpath plugin actually uses the 
+>simple golang `os.MkdirAll` for example when creating mount access based volumes.  Can we do that in all our supported
+>scenarios?  We've also received suggestions around tmpfs or read only bind-mounts of a directory from the CSI driver.
+>**This question most likely can be deferred until implementation.**
+> 3. SELinux implications will need to be properly understood.  Minimally, a consuming Pod needs the correct SELinux 
+>context and fs permissions with bind-mounts for example.
+>**This question most likely can be deferred until implementation.**
+> 4. The EP document in https://github.com/openshift/enhancements/pull/169 needs to be reconciled with this document in
+>some form or fashion.
+
+## Summary
+
+This proposal will describe a [CSI driver](https://github.com/container-storage-interface/spec/blob/master/spec.md) based
+means for OpenShift cluster administrators to share `Secrets` / `ConfigMaps` defined in one namespace with
+other namesapces. 
+
+Any owners/authors/users of Pods in those namespaces can opt into the consumption of the content of those shared `Secrets` / 
+ `ConfigMaps` during their Pod's execution.
+ 
+The precise means of opting in consuming the shared content are outside the scope of this proposal.
+
+This proposal was broken off from [a proposal on injecting RHEL entitlements/subscriptions](https://github.com/openshift/enhancements/pull/214)
+**EDITORIAL NOTE:** if/when PR 214 merges the reference link will be adjusted to point to the merged document hosted on
+https://github.com/openshift/enhancements.  Or if that PR is replaced by another, this proposal will point to that.
+
+
+## Motivation
+
+Simplify consumption of `Secrets` and `ConfigMaps` defined in one namespace in additional namespaces.
+
+### Goals
+
+Provide easy to use sharing of `Secrets` and `ConfigMaps` defined in one namespace in additional namespaces.
+
+### Non-Goals
+
+1. This proposal assumes the sensitive content encapsulated in the `Secret` or `ConfigMap` is valid and correct for the user's purposes.
+No validation of the content for specific usage within a Pod will be done.  That said, this proposal does provide 
+for watching for updates to those objects.
+So if the cluster administrator realized on their own they have provided bad data, and then corrects it, this proposal
+will watch for, discover, and propagate the update when new Pods consuming the content are provisioned.
+
+2. On protecting this content from being viewed/copied by users: if it's available for the Pod to use, the user who can 
+create a Pod that can utilize this feature can also read / view / exfiltrate the Secret contents. It is technically 
+impossible to avoid that.
+
```

Comment:
one more non-goal:  Providing security constraints on who can create these volumes/use them in their pods to access content from another namespace.  Security restrictions are being added/addressed in the related work, not here.

Reply:
part of next push